hydra-node/test/Hydra/NetworkSpec.hs

```diff
@@ -34,61 +34,78 @@ import Test.QuickCheck (
  )
 import Test.QuickCheck.Gen (Gen)
 import Test.QuickCheck.Instances.ByteString ()
-import Test.Util (arbitraryNatural, failAfter, showLogsOnFailure)
+import Test.Util (arbitraryNatural, failAfter)
 
 spec :: Spec
 spec = describe "Networking layer" $ do
-  let requestTx :: HydraMessage SimpleTx
-      requestTx = ReqTx (SimpleTx 1 (utxoRef 1) (utxoRef 2))
-
-      lo = "127.0.0.1"
+  let lo = "127.0.0.1"
 
   describe "Ouroboros Network" $ do
     it "broadcasts messages to single connected peer" $ do
-      received <- newEmptyTMVarIO
-      showLogsOnFailure $ \tracer -> failAfter 10 $ do
-        withOuroborosNetwork tracer (lo, 45778) [(lo, 45779)] (const @_ @(HydraMessage SimpleTx) $ pure ()) $ \hn1 ->
-          withOuroborosNetwork @(HydraMessage SimpleTx) tracer (lo, 45779) [(lo, 45778)] (atomically . putTMVar received) $ \_ -> do
-            broadcast hn1 requestTx
-            atomically (takeTMVar received) `shouldReturn` requestTx
+      received <- atomically newTQueue
+      showLogsOnFailure $ \tracer -> failAfter 30 $ do
+        [port1, port2] <- fmap fromIntegral <$> randomUnusedTCPPorts 2
+        withOuroborosNetwork tracer (lo, port1) [(lo, port2)] (const @_ @Integer $ pure ()) $ \hn1 ->
+          withOuroborosNetwork @Integer tracer (lo, port2) [(lo, port1)] (atomically . writeTQueue received) $ \_ -> do
+            withAsync (1 `broadcastFrom` hn1) $ \_ ->
```

Comment:
Why the async here :thinking: ?

Reply:
I think we still need to broadcast in parallel with the receiving of the message, just in case it takes time for the nodes to (re)connect.